app/controllers/users_controller.rb

```diff
@@ -67,28 +67,48 @@ def list_pending_requested
     @roles = Role.all
   end
 
+
+  # Differences between show _selection and show:
+  #   The key difference between show_selection and the show methods is in how they determine whether the current user has the authority to delete/edit a selected user
+  #   The show method only checks if the current user is not a student or him/herself. It assumes that all other roles are authorised to edit/delete the information of all users.
+  #   The show_selection method allows access only if the current user is higher up the hierarchy of roles than the user s/he is requesting to edit.
+  # The show method additionally calculates the number of total users and the number of groups the user is participating in.
+  # These numbers are used in the show view to display appropriate warning messages when deleting a user.
+
+  # determines if the current user is authorised to see/edit the information of the user in params.
+  # the test used to determine is whether the current user is higher up the hierarchy of roles than the user s/he is requesting to edit.
+  # If these permissions check out, the user is redirected to the 'show' view
+  # If these checks come out negative, the user to given an error message and redirected to a list of all users.
   def show_selection
     @user = User.find_by(name: params[:user][:name])
+    #make sure that the requested user exists
     if !@user.nil?
-      get_role
+      role
+      # if the role is higher in the hierarchy, permission is granted
       if @role.parent_id.nil? || @role.parent_id < session[:user].role_id || @user.id == session[:user].id
         render action: 'show'
       else
+        #else redirect with an error message
         flash[:note] = 'The specified user is not available for editing.'
         redirect_to action: 'list'
       end
+      #redirect to the list of users if the user does not exist.
     else
       flash[:note] = params[:user][:name] + ' does not exist.'
       redirect_to action: 'list'
     end
   end
 
+
+  #finds out the current user's role. If that is not a a student, permission is granted to edit the information of the requested user.
```

Comment:
Following up with my comment on lines 71-76, it may be useful to also indicate where this method is called from (in order to highlight its differences with the `show_selection` method).

Reply:
This has been fixed